awscli/compat.py

```diff
@@ -68,10 +70,15 @@ def get_stdout_text_writer():
         # unicode to the returned stream.  Note that get_stdout_text_writer
         # just returns sys.stdout in the PY3 section above because python3
         # handles this.
-        return codecs.getwriter(locale.getpreferredencoding())(sys.stdout)
+        encoding = locale.getpreferredencoding()
+        if encoding == '':
+            encoding = locale.getpreferredencoding(do_setlocale=False)
+        return codecs.getwriter(encoding)(sys.stdout)
 
     def compat_open(filename, mode='r', encoding=None):
         # See docstring for compat_open in the PY3 section above.
         if 'b' not in mode:
-            encoding = locale.getpreferredencoding()
+            encoding = locale.getpreferredencoding(do_setlocale=False)
```

Comment:
I believe that is what breaks it on Mac OS X.  The default is True, which returns a blank encoding causing the issue.  If it is set to False, then it works.  Perhaps it can be wrapped in an if like lines 73-75.  If I still have the code I'll do a new commit.